### PR TITLE
[xharness] Convert to .NET.

### PR DESCRIPTION
**Pull request by rolfbjarne**
* Convert xharness.csproj and Xharness.Tests.csproj to .NET/sdk-style projects.
* Fix numerous nullability issues that came up.
* Adjust Make logic to do the correct thing now that the executable is named differently.
* Port usage of WebClient to HttpClient, since WebClient is deprecated.
* Find an alternative solution to System.Web.MimeMapping.GetMimeMapping, which
  doesn’t exist in .NET.
* Fix misc other warnings and errors.

### CODE REVIEW

**Review by haritha-mohan — APPROVED** [review]

**Review by dalexsoto — APPROVED** [review]
❤️

**Review by mandel-macaque — APPROVED** [review]

File: tests/xharness/GitHub.cs [hunk 1]
```diff
@@ -99,69 +113,72 @@ IEnumerable<string> GetModifiedFilesRemotely (int pullRequest)
...
-					var rv = new List<string> ();
-					var url = $"https://api.github.com/repos/xamarin/xamarin-macios/pulls/{pullRequest}/files?per_page=100"; // 100 items per page is max
-					do {
-						byte [] data;
-						try {
-							data = client.DownloadData (url);
-						} catch (WebException we) {
-							harness.Log ("Could not load pull request files: {0}\n{1}", we, new StreamReader (we.Response.GetResponseStream ()).ReadToEnd ());
+				var rv = new List<string> ();
+				var url = $"https://api.github.com/repos/xamarin/xamarin-macios/pulls/{pullRequest}/files?per_page=100"; // 100 items per page is max
```
Comment (mcumming) [inline tests/xharness/GitHub.cs:117]: make "https://api.github.com/repos/xamarin/xamarin-macios/" a const

File: tests/xharness/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -4,20 +4,11 @@
 // Information about this assembly is defined by the following attributes.
 // Change them to the values specific to your project.
 
-[assembly: AssemblyTitle ("xharness")]
 [assembly: AssemblyDescription ("")]
-[assembly: AssemblyConfiguration ("")]
-[assembly: AssemblyCompany ("Xamarin")]
-[assembly: AssemblyProduct ("")]
 [assembly: AssemblyCopyright ("(C) Rolf Bjarne Kvinge")]
```
Comment (mcumming) [inline tests/xharness/Properties/AssemblyInfo.cs:8]: Don't think this is correct

**Review by mcumming — APPROVED** [review]
Small things

File: tests/xharness/Properties/AssemblyInfo.cs (hunk 2 above)
Comment (rolfbjarne) [reply tests/xharness/Properties/AssemblyInfo.cs:8]: hehe, this was a few years ago...

**Pull request merged**